force-app/main/default/classes/ListDonation.cls

```diff
@@ -0,0 +1,215 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Donation
+* @description Class for list donation history by credit type.
+*/
+public with sharing class ListDonation {
+
+    static final String HARD_CREDIT = 'Hard Credit';
+    static final String SOFT_CREDIT = 'Soft Credit';
+    static final String PARTIAL_SOFT_CREDIT = 'Partial Soft Credit';
+
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description Donation History inner class.
+    */
+    public class DonationHistory{
+        @AuraEnabled public Id id;
+        @AuraEnabled public String name;
+        @AuraEnabled public Boolean isPrimary;
+        @AuraEnabled public Date closeDate;
+        @AuraEnabled public String role;
+        @AuraEnabled public String creditType;
+        @AuraEnabled public Decimal amount;
+        @AuraEnabled public String paymentMethod;
+    }
+
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description method to return Donation History list.
+    */
+    @AuraEnabled(cacheable=true)
+    public static List<DonationHistory> getDonationHistory(Id contactId, Integer offset){
+        List<OpportunityContactRole> donations = [SELECT Id, Opportunity.Name, Opportunity.CloseDate, Opportunity.Amount, Contact.Email, Contact.Name, Role, isPrimary 
+                                                    FROM OpportunityContactRole 
+                                                    WHERE Opportunity.IsWon = true 
+                                                    AND ContactId =:contactId 
+                                                    ORDER BY Opportunity.CloseDate 
+                                                    DESC LIMIT 50 OFFSET :offset];
```

Comment:
Please keep the [max of 2000 rows](https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_offset.htm) in mind when using `OFFSET`. This may require a "known limitation" statement that the page cannot show more than 2000 gifts or some extra logic to prevent a crash. It's probably unlikely that a single donor would have 2000 related OCR records, but it's not impossible either. Would be good to alert Ali as well.